flood_adapt/integrator/sfincs_adapter.py

```diff
@@ -268,152 +784,17 @@ def add_dis_bc(self, list_df: pd.DataFrame, site_river: list):
-            for aggr_dict in self.site.attrs.fiat.aggregation:
-                # check which one is used in measure
-                if not aggr_dict.name == green_infrastructure.aggregation_area_type:
-                    continue
-                # load geodataframe
-                aggr_areas = gpd.read_file(
-                    measure_path.parents[2] / "static" / "site" / aggr_dict.file,
-                    engine="pyogrio",
-                ).to_crs(4326)
-                # keep only aggregation area chosen
-                polygon_file = aggr_areas.loc[
-                    aggr_areas[aggr_dict.field_name]
-                    == green_infrastructure.aggregation_area_name,
-                    ["geometry"],
-                ].reset_index(drop=True)
-        else:
-            raise ValueError(
-                f"The selection type: {green_infrastructure.selection_type} is not valid"
-            )
-
-        gdf_green_infra = self.sf_model.data_catalog.get_geodataframe(
-            polygon_file,
-            geom=self.sf_model.region,
-            crs=self.sf_model.crs,
-        )
-
-        # Make sure no multipolygons are there
-        gdf_green_infra = gdf_green_infra.explode()
-
-        # Volume is always already calculated and is converted to m3 for SFINCS
-        height = None
-        volume = green_infrastructure.volume.convert(UnitTypesVolume("m3"))
-
-        # HydroMT function: create storage volume
-        self.sf_model.setup_storage_volume(
-            storage_locs=gdf_green_infra, volume=volume, height=height, merge=True
-        )
-
-    def add_pump(self, pump: PumpModel, measure_path: Path):
-        """Add pump to sfincs model.
-
-        Parameters
-        ----------
-        pump : PumpModel
-            pump information
-        """
-        # HydroMT function: get geodataframe from filename
-        polygon_file = measure_path.joinpath(pump.polygon_file)
-        gdf_pump = self.sf_model.data_catalog.get_geodataframe(
-            polygon_file, geom=self.sf_model.region, crs=self.sf_model.crs
-        )
-
-        # HydroMT function: create floodwall
-        self.sf_model.setup_drainage_structures(
-            structures=gdf_pump,
-            stype="pump",
-            discharge=pump.discharge.convert(UnitTypesDischarge("m3/s")),
-            merge=True,
-        )
-
-    def write_sfincs_model(self, path_out: Path):
-        """Write all the files for the sfincs model.
-
-        Args:
-            path_out (Path): new root of sfincs model
-        """
-        # Change model root to new folder
-        self.sf_model.set_root(path_out, mode="w+")
-
-        # Write sfincs files in output folder
-        self.sf_model.write()
-
-    def add_spw_forcing(
+    def _add_forcing_spw(
         self,
         historical_hurricane: HistoricalHurricane,
         database_path: Path,
         model_dir: Path,
     ):
+        # TODO investigate this. seems to not add any forcing?
```

Comment:
there is a function further up that adds the keyword spwfile to the sfincs input file. that should probably be included in here